src/Wait.hs

```diff
@@ -20,13 +22,14 @@ import System.FSNotify
 import Language.Haskell.Ghcid.Util
 
 
-data Waiter = WaiterPoll Seconds
-            | WaiterNotify WatchManager (MVar ()) (Var (Map.Map FilePath StopListening))
+data Waiter a
+  = WaiterPoll Seconds
+  | WaiterNotify WatchManager (MVar ()) (Var (Map.Map (FilePath, a) StopListening))
```

Comment:
Do we need the `a` in the `Map` at all? The WaiterNotify just records which watches we have out with the underlying inotify watcher. We only use the WaiterNotify to kick us if something happens, therefore I think removing the `a` here is safe, since we use the `recheck` loop to raise the right things. Does that make sense?

Reply:
You're right, we don't! I've removed it.